File: fabric-starter/chaincode/go/shipment/shipment.go
package main

import (
	"time"
)

type Shipment struct {
	TrackingCode        string    `json:"trackingCode"`
	Courier             string    `json:"courier"`
	Sender              Sender    `json:"sender"`
	Recipient           Recipient `json:"recipient"`
	WeightInGrams       int       `json:"weightInGrams"`
	Size
	Content             string    `json:"content"`
	ShippingType        string    `json:"shippingType"`
	IsFragile           bool      `json:"isFragile"`
	LastState           string    `json:"lastState"`
	LastLocation        *Address  `json:"lastLocation"`
	IsInCourierFacility bool      `json:"isInCourierFacility"`
	CreatedAt           time.Time `json:"createdAt"`
	UpdatedAt           time.Time `json:"updatedAt"`
}

type ShipmentRecord struct {
	Key   string `json:"key"`
	Value string `json:"value"`
}

type Shipments []ShipmentRecord

func NewShipment(
	trackingCode string,
	courier string,
	sender Sender,
	recipient Recipient,
	weightInGrams int,
	shippingType string,
	size Size,
	content string,
	isFragile bool,
	lastState string,
	createdAt time.Time,
	updatedAt time.Time,
) *Shipment {
	shipment := &Shipment{
		TrackingCode:        trackingCode,
		Courier:             courier,
		Sender:              sender,
		Recipient:           recipient,
		WeightInGrams:       weightInGrams,
		ShippingType:        shippingType,
		Size:                size,
		Content:             content,
		IsFragile:           isFragile,
		LastState:           lastState,
		IsInCourierFacility: false,
		CreatedAt:           createdAt,
		UpdatedAt:           updatedAt,
	}

	return shipment
}

File: courier/web/src/app/core/models/shipment.ts
export interface Shipment {
  trackingCode: string;
  courier: string;
  shippingType: string;
  sender: {
    firstName: string;
    lastName: string;
    country: string;
    city: string;
    postCode: string;
    address: string;
    phone: string;
  },
  recipient: {
    firstName: string;
    lastName: string;
    country: string;
    city: string;
    postCode: string;
    address: string;
    phone: string;
  },
  weightInGrams: number;
  heightInMM: number;
  widthInMM: number;
  depthInMM: number;
  content: string;
  isFragile: boolean;
  lastState: string;
  lastLocation?: {
    country: string;
    city: string;
    postCode: string;
    address: string;
  };
  isDelivered: boolean;
  createdAt: Date;
  updatedAt: Date;
}
